webapp/src/components/Modals/BuyWithCryptoModal/utils.ts

```diff
@@ -207,81 +201,50 @@ export const getDefaultChains = () => {
   asset: Item
-): Promise<Route> =>
-  crossChainProvider.getMintNFTRoute({
-    ...baseRouteConfig,
-    item: {
-      collectionAddress: asset.contractAddress,
-      itemId: asset.itemId,
-      price: asset.price
-    },
-    toAmount: asset.price,
-    toChain: asset.chainId
-  })
+): Promise<BigNumber> => {
+  const networkProvider = await getNetworkProvider(selectedChain)
+  const provider = new ethers.providers.Web3Provider(networkProvider)
+
+  const contract = getContract(ContractName.CollectionStore, asset.chainId)
+  const c = new ethers.Contract(contract.address, contract.abi, provider)
+  return c.estimateGas.buy(
+    [[asset.contractAddress, [asset.itemId], [asset.price], [wallet.address]]],
+    { from: wallet.address }
+  )
+}
 
-export const estimateTransactionGas = async (
+export const estimateBuyNftGas = async (
   selectedChain: ChainId,
   wallet: Wallet,
-  asset: Asset,
-  order?: Order
-) => {
+  asset: NFT,
+  order: Order
+): Promise<BigNumber> => {
   const networkProvider = await getNetworkProvider(selectedChain)
   const provider = new ethers.providers.Web3Provider(networkProvider)
 
-  let estimation: BigNumber | undefined
-  if (order && isNFT(asset)) {
-    const contractName = getContractName(order.marketplaceAddress)
-    const contract = getContract(contractName, order.chainId)
-    const c = new ethers.Contract(contract.address, contract.abi, provider)
-    estimation = await c.estimateGas.executeOrder(
-      asset.contractAddress,
-      asset.tokenId,
-      order.price,
-      { from: wallet.address }
-    )
-  } else if (!isNFT(asset)) {
-    const contract = getContract(ContractName.CollectionStore, asset.chainId)
-    const c = new ethers.Contract(contract.address, contract.abi, provider)
-    estimation = await c.estimateGas.buy(
-      [
-        [asset.contractAddress, [asset.itemId], [asset.price], [wallet.address]]
-      ],
-      { from: wallet.address }
-    )
-  }
-  return estimation
+  const contractName = getContractName(order.marketplaceAddress)
+  const contract = getContract(contractName, order.chainId)
+  const c = new ethers.Contract(contract.address, contract.abi, provider)
+  return c.estimateGas.executeOrder(
+    asset.contractAddress,
+    asset.tokenId,
+    order.price,
+    { from: wallet.address }
+  )
 }
 
 export const estimateNameMintingGas = async (
   name: string,
   selectedChain: ChainId,
   ownerAddress: string
-) => {
+): Promise<BigNumber> => {
   const networkProvider = await getNetworkProvider(selectedChain)
   const provider = new ethers.providers.Web3Provider(networkProvider)
-
-  const contract = getContract(ContractName.DCLRegistrar, selectedChain)
+  const contract = getContract(ContractName.DCLControllerV2, selectedChain)
   const c = new ethers.Contract(contract.address, contract.abi, provider)
-  const estimation = await c.estimateGas.register(ownerAddress, name)
-  return estimation
+  return c.estimateGas.register(name, ownerAddress)
```

Comment:
```suggestion
  return c.estimateGas.register(ownerAddress, name)
```
I belive this should remaing like this, is there any reason why we're changing it?

Reply:
I've changed the contract, from the `DCLRegistrar` to the `DCLControllverV2` because it was wrong and the `register` method to have [the correct parameters according to its ABI](https://github.com/decentraland/decentraland-transactions/blob/6d4a6d59b3cd5f0961308f9492e62ecdcc12d88e/src/abis/DCLControllerV2.ts#L134) which have the name first and then the address of the beneficiary.